src/creds.rs

```diff
@@ -173,16 +173,16 @@ impl Codec for BasicCredential {
         self.public_key.encode(buffer)?;
         Ok(())
     }
-    // fn decode(cursor: &mut Cursor) -> Result<Self, CodecError> {
-    //     let identity = decode_vec(VecSize::VecU16, cursor)?;
-    //     let ciphersuite = Ciphersuite::decode(cursor)?;
-    //     let public_key = SignaturePublicKey::decode(cursor)?;
-    //     Ok(BasicCredential {
-    //         identity,
-    //         ciphersuite,
-    //         public_key,
-    //     })
-    // }
+    fn decode(cursor: &mut Cursor) -> Result<Self, CodecError> {
+        let identity = decode_vec(VecSize::VecU16, cursor)?;
+        let ciphersuite = Ciphersuite::decode(cursor)?;
+        let public_key = SignaturePublicKey::decode(cursor)?;
+        Ok(BasicCredential {
+            identity,
+            ciphersuite,
+            public_key,
+        })
+    }
```

Comment:
Would be nice to have encoding/decoding tests for both `Credential` and `BasicCredential`

Reply:
I agree, I filed #130 to write tests for encoding and decoding in general. This specific case might also get covered before when we look at credentials.